app/src/main/java/app/suhasdissa/vibeyou/ui/screens/player/EqualizerSheet.kt

```diff
@@ -0,0 +1,198 @@
+                                            text = option.capitalize(Locale.current),
+                                            style = MaterialTheme.typography.bodyLarge
+                                        )
+                                    },
+                                    onClick = {
+                                        selectedPresetIndex = index
+
+                                        Pref.sharedPreferences.edit {
+                                            putInt(Pref.equalizerPresetKey, index - 1)
+                                        }
+                                        playerViewModel.updateEqualizerSettings()
+
+                                        expanded = false
+                                    },
+                                    contentPadding = ExposedDropdownMenuDefaults.ItemContentPadding,
+                                )
+                            }
+                        }
+                    }
+                }
+
+                Spacer(modifier = Modifier.height(20.dp))
+
+                AnimatedVisibility(visible = selectedPresetIndex == 0) {
+                    Column {
+                        val bandLevels = remember {
+                            val list = Pref.sharedPreferences.getString(Pref.equalizerBandsKey, "")
+                                .takeIf { !it.isNullOrEmpty() }?.split(",")?.map(String::toShort)
+                            // init at 0 - default/normal band volume
+                                ?: List(equalizerData.bands.size) { (0).toShort() }
+
+                            list.toMutableStateList()
+                        }
+
+                        fun onUpdateBands() {
+                            Pref.sharedPreferences.edit {
+                                putString(
+                                    Pref.equalizerBandsKey, bandLevels.joinToString(",")
+                                )
+                            }
+                            playerViewModel.updateEqualizerSettings()
+                        }
+
+                        Row(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween
+                        ) {
+                            equalizerData.bands.forEachIndexed { index, band ->
+                                Column(
+                                    horizontalAlignment = Alignment.CenterHorizontally
+                                ) {
+                                    VerticalSlider(
+                                        modifier = Modifier.width(300.dp),
+                                        value = bandLevels[index].toFloat(),
+                                        onValueChange = {
+                                            bandLevels[index] = it.toInt().toShort()
+                                            onUpdateBands()
+                                        },
```

Comment:
I don't think it's efficient to update shared preferences while dragging the slider.

```suggestion
                                        onValueChange = {
                                            // Update equalizer settings (without saving shared preferences)
                                        },
                                        onValueChangeFinished = {
                                           // Save preferences
                                        },
```